sentinel/test/mocha/transitions/update_notifications.js

```diff
@@ -0,0 +1,355 @@
+const sinon = require('sinon').sandbox.create(),
+      transition = require('../../../transitions/update_notifications'),
+      db = require('../../../db-nano'),
+      utils = require('../../../lib/utils');
+
+describe('update_notifications', () => {
+
+  afterEach(done => {
+    sinon.restore();
+    done();
+  });
+
+  describe('filter', () => {
+
+    it('empty doc does not match', () => {
+      transition.filter({}).should.equal(false);
+    });
+
+    it('missing form does not match', () => {
+      transition.filter({
+        fields: { patient_id: 'x' }
+      }).should.equal(false);
+    });
+
+    it('missing clinic phone does not match', () => {
+      transition.filter({
+        form: 'x',
+        fields: { patient_id: 'x' }
+      }).should.equal(false);
+    });
+
+    it('already run does not match', () => {
+      transition.filter({
+        form: 'x',
+        fields: { patient_id: 'x' },
+        transitions: {
+          update_notifications: { last_rev: 9, seq: 1854, ok: true }
+        }
+      }).should.equal(false);
+    });
+
+    it('match', () => {
+      transition.filter({
+        form: 'x',
+        fields: { patient_id: 'x' },
+        type: 'data_record'
+      }).should.equal(true);
+    });
+
+  });
+
+  describe('onMatch', () => {
+
+    it('returns false if not on or off form', done => {
+      sinon.stub(transition, 'getConfig').returns({
+        on_form: 'x',
+        off_form: 'y'
+      });
+      const change = {
+        doc: {
+          form: 'z',
+          type: 'data_record'
+        }
+      };
+      transition.onMatch(change).then(changed => {
+        (!!changed).should.equal(false);
+        done();
```

Comment:
Also not sure if you're aware but you can return a promise from a test and it'll wait for it to complete instead of using `done`

Reply:
Yeah and I much prefer that. Good spotting, thanks!